src/lib/global-module.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  ClassProvider,
+  DynamicModule,
+  ExistingProvider,
+  FactoryProvider,
+  Global,
+  LoggerService,
+  Module,
+  ValueProvider,
+} from '@nestjs/common';
+import { Type } from '@nestjs/common/interfaces/type.interface';
+import { ForwardReference } from '@nestjs/common/interfaces/modules/forward-reference.interface';
+import * as dotenv from 'dotenv';
+import { ConfigOptions } from '../options';
+import { ConfigFacade } from './facade';
+import { ConfigExtractor } from './extractor';
+import { ConfigParser } from './parser';
+import { ConfigFactory } from './factory';
+import { ConfigValidator } from './validator';
+import { CONFIG_LOGGER, CONFIG_OPTIONS } from '../tokens';
+
+@Global()
+@Module({
+  providers: [
+    {
+      provide: ConfigExtractor,
+      useFactory: () => new ConfigExtractor(dotenv.config),
+    },
+    {
+      provide: ConfigParser,
+      useClass: ConfigParser,
+    },
+    {
+      provide: ConfigFactory,
+      useClass: ConfigFactory,
+    },
+    {
+      provide: ConfigValidator,
+      useClass: ConfigValidator,
+    },
+  ],
+})
+export class ConfigGlobalModule {
+  static forRoot(options: ConfigOptions): DynamicModule {
+    const imports: Array<
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      Type<any> | DynamicModule | Promise<DynamicModule> | ForwardReference
+    > = options.imports || [];
+    const providers: Array<
+      ClassProvider | FactoryProvider | ValueProvider | ExistingProvider
+    > = options.providers || [];
+
+    if (!providers.find((provider) => provider.provide === CONFIG_LOGGER)) {
+      providers.push({
+        provide: CONFIG_LOGGER,
+        useValue: console,
```

Comment:
it's bad idea to use `console` as a logger. If there is no logger in providers we just don't logging it

Reply:
we can create a wrapper provider that depends on this logger. And there decide log or not. It will help us not to change `this.logger....`